app/models/manageiq/providers/azure/cloud_manager/provision/cloning.rb

```diff
@@ -82,13 +80,22 @@ def prepare_for_clone_task
             :caching      => 'ReadWrite',
             :osType       => os
           }
-        },
-        :networkProfile  => {
-          :networkInterfaces => [{:id => nic_id}],
         }
       }
     }
 
+    # The -1 value is set in ProvisionWorkflow to distinguish between the
+    # desire for a new Public IP address vs a private IP.
+    #
+    if floating_ip
+      nic_id = associated_nic || create_nic(true)
+    else
+      public_ip = options[:floating_ip_address].first == -1
```

Comment:
@djberg96 - Can you point me to where the options hash gets populated? What does this line mean?

Reply:
It's coming from the UI. The line in question is https://github.com/djberg96/manageiq-providers-azure/blob/70fbc2f62b293ad1490a1530b98ae28473e6632a/app/models/manageiq/providers/azure/cloud_manager/provision_workflow.rb#L30

Reply:
I was looking for the code that populates the options hash with all the valid floating Ips.
Does this line return what the user selecting from the floating_ip drop down?

Reply:
I'm not sure where it's populated originally, somewhere on the UI side. Yes, that line returns what the user selected as a key/value pair on a typical Rails dropdown menu. I interjected the -1 there for us to distinguish between "create new" and "not selected".

I would have preferred to use a string key to be more descriptive, but the UI code blew up when it tried to sort it.

Reply:
So the reason I was looking for the code that populates the options hash is because this line returns an array of floating ips, so its a multi-select? Will `first` always be -1?

Reply:
Oops I misspoke. The `options[:floating_ip_address]` will be a two element array of the currently selected item, i.e. [-1, "New"]. If the user actually selects something, then it will be something else.